bounded_counter/count_min_sketch.py

```diff
@@ -0,0 +1,99 @@
+import os
+import numpy
+import xxhash
+
+
+def hash_key(i, key):
+    return xxhash.xxh64(str(key), seed=i).intdigest()
+
+algorithm_basic = 0
+algorithm_conservative = 1
+algorithm_logcounter = 2
+
+class CountMinSketch(object):
+    def __init__(self, width=16, depth=8, algorithm = None):
+        self.logshift = 0
+        array_type = numpy.uint32
+        self.algorithm = algorithm_basic
+        self.inc_method = self.basic_increment
+        if algorithm and str(algorithm).lower() == 'conservative':
+            self.algorithm = algorithm_conservative
+            self.inc_method = self.conservative_increment
+        if algorithm and str(algorithm).lower() == 'logcounter8':
+            array_type = numpy.uint8
+            self.algorithm = algorithm_logcounter
+            self.logshift = 2
+            self.inc_method = self.log_increment
+        if algorithm and str(algorithm).lower() == 'logcounter1024':
+            array_type = numpy.uint16
+            self.algorithm = algorithm_logcounter
+            self.logshift = 9
+            self.inc_method = self.log_increment
+
+        self.width = width
+        self.depth = depth
+        self.hash_length = width.bit_length()
+        self.hash_mask = (1 << (self.hash_length - 1)) - 1
+        self.logbase = 1 << (self.logshift + 1)
+        self.table = numpy.zeros((depth, width), array_type)
+
+
+    def bucket(self, i, key):
+        return self.hash_mask & hash_key(i, key)
+
+    def increment(self, key):
+        self.inc_method(key)
+
+    def basic_increment(self, key):
+        for i in range(self.depth):
```

Comment:
`xrange` more efficient (generator vs an explicit list). Here and elsewhere.

Reply:
I'm using python 3 where xrange == range. Should I switch to 2 instead?

Reply:
Yes, we support both (2.7+ and 3.4+), using the [six](https://pythonhosted.org/six/) library, the differences are minimal.

The exception would be if we had to use some Python3-specific features (`async`?), but I don't think that's needed here.